frontend/simple/views/components/Voting.vue

```diff
@@ -0,0 +1,292 @@
+import { toPercent } from '../utils/filters'
+
+export default {
+  name: 'Voting',
+  props: {
+    proposal: Object
+  },
+  components: {
+    ButtonCountdown
+  },
+  data () {
+    return {
+      closeProposalState: {
+        // opts from ButtonCountdown
+      }
+    }
+  },
+  computed: {
+    svgCircle () {
+      const svgCircleP = 220 // 35*2 * 3.14
+      const ruleVal = this.proposal.value
+      const ruleWarning = 0.7
+
+      return {
+        style: {
+          strokeDasharray: `${svgCircleP * ruleVal} ${svgCircleP}`
+        },
+        class: ruleVal < ruleWarning && 'has-stroke-warning'
+      }
+    },
+    isTypeMember () {
+      return this.proposal.type === 'member'
+    },
+    isTypeRule () {
+      return this.proposal.type === 'rule'
+    },
+    isTypeMincome () {
+      return this.proposal.type === 'mincome'
+    },
+    isTypeRuleOrMincome () {
+      return ['rule', 'mincome'].includes(this.proposal.type)
+    },
+    hasVoted () {
+      return this.proposal.userVote !== null
+    },
+    hasVotedFor () {
+      return this.proposal.userVote
+    },
+    hasVotedAgainst () {
+      return this.proposal.userVote === false
+    },
+    proposalValue () {
+      return this.isTypeRule ? toPercent(this.proposal.value) : this.proposal.value
+    },
+    isProposalClosed () {
+      return this.closeProposalState.state === 'success'
+    },
+    closeProposalBtnText () {
+      return this.closeProposalState.state === 'counting'
+        ? Lang('No, wait')
+        : Lang('Close Proposal')
+    },
+    helperText () {
+      switch (this.closeProposalState.state) {
+        case 'counting':
+          return template(
+            Lang('Proposal closed in {countdown}...'), {
+              countdown: this.closeProposalState.countdown
+            })
+        case 'cancelled':
+          return Lang("Let's pretend that never happened")
+        case 'success':
+          return Lang('Proposal cancelled')
+        default:
+          break
+      }
+
+      if (!this.proposal.votes) {
+        return this.proposal.ownProposal
+          ? Lang('Nobody voted yet')
+          : Lang('Be the first to vote!')
+      }
+
+      if (!this.proposal.ownProposal && this.proposal.votes === 1) {
+        return Lang('Your were the first to vote')
+      }
+
+      if (this.proposal.votes > 1) {
+        return template(Lang('{youAnd} {votesCount} of {members} members voted'), {
+          youAnd: this.hasVoted ? Lang('You and') : '',
+          votesCount: this.hasVoted ? this.proposal.votes - 1 : this.proposal.votes,
+          members: this.proposal.members
+        })
+      }
+    }
+  },
+  methods: {
+    handleCloseProposalStateChange (state, opts = {}) {
+      this.closeProposalState = { state, ...opts }
+
+      state === 'success' && this.closeProposal()
+    },
+    handleVoteAgainst () {
+      if (this.hasVotedAgainst) {
+        return false
+      }
+      console.log('TODO Logic - The user voted against')
```

Comment:
this could call a function passed down as a prop from the connected container, that way we could move the todo there and this component could be 100% ready :)

Reply:
Great suggestion, done!